src/sqlfluff/rules/L039.py

```diff
@@ -66,4 +66,22 @@ def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
                         )
                 prev_newline = False
                 prev_whitespace = None
+
+            if seg.is_type("object_reference"):
+                # This variable is a workaround to avoid removing new indents added at the beginning of a segment by L003.
+                # See Github issue #1304: https://github.com/sqlfluff/sqlfluff/issues/1304
+                # It represents the question: are we parsing through leading whitespace in this loop?
+                leading_whitespace = True
```

Comment:
@jpy-git adding this variable to track whether we're still in leading whitespace is my proposed approach.